From fccc2d31a3d18046985e8ff070fa851d8d816b24 Mon Sep 17 00:00:00 2001
From: baltevl <baltevl@disroot.org>
Date: Fri, 5 Sep 2025 09:03:58 +0200
Subject: [PATCH 097/133] std.http.Client.Connection: make host() public Closes
 #25153
X-Developer-Signature: v=1; a=openpgp-sha256; l=637; i=xtex@astrafall.org;
 h=from:subject; bh=X/9H2LlVBZWuRumJP9FYrhu5fJMR+6MnjTSlNBW51uI=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78ri6zB84b/Y9XLb2sevJTxqMc1ee5yzu49pfveN8/
 90kj4CyjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpfhPG8D/LSFamXGvxr7+LCnQCDTJ+BwhbTN+5Mcwse5v87sumE8wY/pdLLyn
 PKA5+ZCP90Tsj5fWkSvnHyhKSycfT2lMfr2JUZgIA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 34409635ad3a5aa199adde1cabbeadb1ac6e256c)
---
 lib/std/http/Client.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/std/http/Client.zig b/lib/std/http/Client.zig
index 37573917359e..808c2a3dba3b 100644
--- a/lib/std/http/Client.zig
+++ b/lib/std/http/Client.zig
@@ -382,7 +382,7 @@ pub const Connection = struct {
         return c.stream_reader.getStream();
     }
 
-    fn host(c: *Connection) []u8 {
+    pub fn host(c: *Connection) []u8 {
         return switch (c.protocol) {
             .tls => {
                 if (disable_tls) unreachable;
-- 
2.52.0

